File: loss/MeanSquaredError.m
classdef MeanSquaredError < BaseLoss
    %MEANSQUAREDERROR Summary of this class goes here
    %   Detailed explanation goes here
    
    properties
    end
    
    methods (Access = 'public')
        function [loss, gradients] = GetLossAndGradients(obj, scores, targets)
            loss = [];
            gradients = [];
        end
    end
    
end

File: utils/optimization_functions/cppPrj/loss/meansquarederror.cpp
#include "meansquarederror.h"

MeanSquaredError::MeanSquaredError()
{

}

tuple<float, Tensor<float> > MeanSquaredError::GetLossAndGradients(const Tensor<float> &scores, const Tensor<float> &targets)
{
    cout << "From MeanSquaredError" << endl;
    return make_tuple(0.1F,Tensor<float>());
}
